src/renderer/pages/project/components/script_editor/beat_preview.vue

```
@@ -1,67 +1,105 @@
+        class="relative border-2 border-dashed border-gray-300 rounded-lg p-4 text-center"
+        :key="`beat_editor_${beat.id ?? index}`"
+      >
+        <!-- Generate image button -->
+        <template v-if="shouldShowGenerateButton">
+          <Button
+            variant="ghost"
+            size="icon"
+            class="absolute -left-3 -top-3 z-10 bg-white border border-gray-300 rounded-full shadow hover:bg-gray-100 transition-colors w-8 h-8 flex items-center justify-center"
+            @click="generateImage"
+            :disabled="isImageGenerating || isHtmlGenerating || props.toggleTypeMode"
+            :title="props.toggleTypeMode ? 'Change beat type first' : (isImageGenerating || isHtmlGenerating) ? 'Generating...' : 'Generate image'"
```

ESLint

[error] Replace `props.toggleTypeMode·?·'Change·beat·type·first'·:·(isImageGenerating·||·isHtmlGenerating)·?·'Generating...'·:·'Generate·image'` with `⏎··············props.toggleTypeMode⏎················?·'Change·beat·type·first'⏎················:·isImageGenerating·||·isHtmlGenerating⏎··················?·'Generating...'⏎··················:·'Generate·image'⏎············`

(prettier/prettier)